### DISCUSSION TITLE
1.0.0-alpha.1

### DISCUSSION THREAD

**Discussion by mxcl** [pkgxdev/pkgx#700] [Announcements]
1.0.0-alpha.1 is a seminal release, we welcome feedback, especially from v0 early adopters.

Release notes: [[pkgx blog]](https://blog.pkgx.dev/pkgx-1-0-0-alpha-1/)

**Comment by davdroman** [comment]
I'm yet to try the release, but reading through the blog post I found myself nodding along at every argument for every change and removal, so that's tremendously exciting.

Congrats to the entire `tea` team for this historic milestone!

**Reply by mxcl** [reply]
Thank you David. I have spent the last 6 weeks living and breathing tea/cli from morning till night and then in my dreams most likely too until the shape of it all consolidated. Though as I said in the post, it was the community’s input that guided us. We truly could not have done it without open source and having been given the leeway and the bandwidth to experiment from our initial release by our early adopters (like yourself!)

You can’t do it that way if you don’t develop in the open.

Genuinely though I think tea/cli as a runner that can run anything is our purpose. A small tool you can reach for in any scenario to achieve your goals with the wealth of tooling in open source. Package managers shape your system, tea enables your work.

**Reply by NickShh1** [reply]
Wow Mxcl! I can see you've done a lot of work! I wanted to point out that open source is an integral part of decentralization and the entire Web 3! Glad to see Tea working for the benefit of users and developers!

**Reply by MAZLOUM88** [reply]
Wow great

**Reply by GabrielZDX** [reply]
I'm very excited to start interacting with testnet

**Reply by Robiansah** [reply]
bandwidth to experiment from gitclone?

**Reply by ibnstwn** [reply]
Let's build it

**Comment by jkhoeini** [comment]
Congrats on the new release. Great work team tea 🎉 

I found `tea` just a few weeks back and it has changed my life 🥳

Fun fact: I found out about the v1 release before the release since I was checking the docs and the new docs were live before the release actually happened :D

I am currently using the README file dependency section and wanted to share with you how I'm using it. In my work, most of my colleagues are less experienced with js tooling. So I've set up a clear README file linking to your home page and a dependencies section. Explaining that using tea is optional as long as they can install the dependencies correctly by themselves. Also in the README I've put tldr script sections on how to setup tea, or how to just use `sh <(curl tea.xyz)` in front of everything. Afterward, they can just do a `tea -S` once in the root folder of the project if they choose tea. Also through your docs, I found out about `xc`. That's another amazing tool that changed my life. Thanks for that as well. It's so great finding out about good tools through your docs. ❤️
So, they can just do `xc dev` and everything will just work. That's unreal to me how great everything works. It will also give perfect freedom and clarity to everyone on how they choose to install DEPs or run the project. We use IntelliJ and it allows my colleagues to just click the play button in front of the script sections in the README. This workflow is one of those 🤯 moments for me that I haven't had the experience of for many years.

Thanks again for the great tool.

That being said I don't want to make the impression that I'm asking for the README deps to be back. I just felt it could be useful to share my usage of it with you. However, I wholeheartedly support the idea of having tea doing one thing and doing it great and instead building tools on top of it. For example, I can imagine a workflow based on `libtea` that I think would be equally mind-blowing and great to me. If I could just have a short `deno` script in the `scripts/` folder of my project that did the same thing I could also just `deno compile` it and put the binary in the git repo as well and point the README instructions to use that instead. For inspiration, I think if we could have something like this it would be perfect:

```ts
import tea from 'libtea';
import mdParser from 'some-awesome-markdown-lib';

const $ = mdParser('README.md');
const depsList = $('#dependencies > table');
const deps = Object.fromEntries(depsList);

tea.install(deps);
```

**Reply by mxcl** [reply]
k so probs we could add README parsing back in. The “perceived complexity of tea” is much reduced IMO by `dev` being a separate tool.

Do you still prefer README over eg. tea.yaml?

Thank you for being an early adopter and thank you for the feedback as it is quite important. In my mind nobody used the README deps feature since I'd heard no users say they did and several users say that they disliked it.

**Reply by trentlarson** [reply]
I used the README. I'm not married to the idea and I'll pull it out but I often need a way to specify multiple tools (eg. python in a node project for node-gyp, and java and ruby in a react-native node project for building android and deploying); the new way looks like pkgx.yaml which is fine and I can link to that in the README for people doing it by hand... where is the best place to look (for more detail than [this](https://docs.pkgx.sh/using-dev/dev#pkgx.yaml))?

**Reply by mxcl** [reply]
well, the suggested way is either `pkgx.yml` or YAML front matter in the project keyfiles. eg. `package.json` or `pyproject.toml`.

What more detail are you looking for?

**Reply by trentlarson** [reply]
I'm looking for the format of pkgx.yaml. Maybe it's just a "dependencies" array, thus:

```
dependencies:
  - deno@1.35
  - ...
```

**Reply by mxcl** [reply]
right, the format is the same as that of YAML front matter. But I will clarify that. Thanks.

edit: https://github.com/pkgxdev/pkgx/commit/d5e5f998c7df4cd2482a2c9984fa0a2bc2798ee7

LMK if you think it needs more. Thank you.

**Reply by trentlarson** [reply]
That's helpful! Thanks.

**Reply by MAZLOUM88** [reply]
Helpful

**Reply by hoakine1** [reply]
Thanks

**Comment by trentlarson** [comment]
Note that your [tea blog] link to this is broken: https://tea.xyz/blog/tea-1-0-0-alpha-1/

**Reply by mxcl** [reply]
Thank you fixed. https://blog.pkgx.dev/pkgx-1-0-0-alpha-1/

**Comment by adam50200** [comment]
Good

**Comment by noyjacker** [comment]
Will tea use this program for the testnet later?

**Comment by Claire3653** [comment]
Congrats on the huge milestone! 
I can't wait to become tea taster !

**Comment by dorikunz** [comment]
good!!

**Comment by Nikaars** [comment]
Great!!

**Comment by Edotamara26** [comment]
That's great for projects 🔥

**Comment by Edotamara26** [comment]
> 1.0.0-alpha.1 is a seminal release, we welcome feedback, especially from v0 early adopters.
> 
> Release notes: [[pkgx blog]](https://blog.pkgx.dev/pkgx-1-0-0-alpha-1/)

Thank for sent information

**Comment by hanriiver** [comment]
\   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||
